go/vt/vttablet/tabletserver/query_engine_test.go

```diff
@@ -266,3 +267,73 @@ func newTestQueryEngine(queryPlanCacheSize int, idleTimeout time.Duration, stric
 	qe.InitDBConfig(dbcfgs)
 	return qe
 }
+
+func runConsolidatedQuery(t *testing.T, sql string) *QueryEngine {
+	db := fakesqldb.New(t)
+	defer db.Close()
+
+	testUtils := newTestUtils()
+	dbcfgs := testUtils.newDBConfigs(db)
+	qe := newTestQueryEngine(10, 1*time.Second, true, dbcfgs)
+	qe.se.Open()
+	qe.Open()
+	defer qe.Close()
+
+	c := make(chan int)
+	done := make(chan int)
+	query := func(sql string, waiter chan int, done chan int) {
+		q, ok := qe.consolidator.Create(string(sql))
+		if ok {
+			q.Result, q.Err = "ok", nil
+			<-waiter
+			q.Broadcast()
+		} else {
+			waiter <- 1
+			q.Wait()
+
+			done <- 1
+		}
+	}
+
+	go query(sql, c, done)
+	go query(sql, c, done)
+
+	<-done
+
+	return qe
+}
+
+func TestConsolidationsUIRedaction(t *testing.T) {
+	// Reset to default redaction state.
+	defer func() {
+		*streamlog.RedactDebugUIQueries = false
+	}()
+
+	request, _ := http.NewRequest("GET", "/debug/consolidations", nil)
+
+	sql := "select * from test_db_01 where column = 'secret'"
+	redactedSQL := "select * from test_db_01 where `column` = :redacted1"
+
+	// First with the redaction off
+	*streamlog.RedactDebugUIQueries = false
+	unRedactedResponse := httptest.NewRecorder()
+	qe := runConsolidatedQuery(t, sql)
+
+	qe.ServeHTTP(unRedactedResponse, request)
+	if !strings.Contains(unRedactedResponse.Body.String(), sql) {
+		t.Fatalf("Response is missing the consolidated query: %v %v", sql, unRedactedResponse.Body.String())
+	}
+
+	// Now with the redaction on
+	*streamlog.RedactDebugUIQueries = true
```

Comment:
This test looks good but you should set this back to false in a defer so that it resets when the test completes (at which point I think the txlogz_test.go change isn't needed, though it's fine for safety).

Reply:
You mean like the defer that's in line 1 of this function 😹